sdk/src/wallet/core/builder.rs

```diff
@@ -134,11 +132,16 @@ where
             }
         }
 
-        #[cfg(all(feature = "rocksdb", feature = "storage"))]
-        let storage =
-            crate::wallet::storage::adapter::rocksdb::RocksdbStorageAdapter::new(storage_options.path.clone())?;
-        #[cfg(all(not(feature = "rocksdb"), feature = "storage"))]
-        let storage = Memory::default();
+        #[cfg(feature = "storage")]
+        let storage = cfg_if::cfg_if!(
+            if #[cfg(feature = "rocksdb")] {
+                crate::wallet::storage::adapter::rocksdb::RocksdbStorageAdapter::new(storage_options.path.clone())?
+            } else if #[cfg(feature = "jammdb")] {
+                crate::wallet::storage::adapter::jammdb::JammdbStorageAdapter::new(storage_options.path.clone())?
+            } else {
+                crate::wallet::storage::adapter::memory::Memory::default()
+            }
+        );
```

Comment:
```suggestion
        cfg_if::cfg_if!(
            if #[cfg(feature = "rocksdb")] {
                let storage = crate::wallet::storage::adapter::rocksdb::RocksdbStorageAdapter::new(storage_options.path.clone())?;
            } else if #[cfg(feature = "jammdb")] {
                let storage = crate::wallet::storage::adapter::jammdb::JammdbStorageAdapter::new(storage_options.path.clone())?;
            } else {
                let storage = crate::wallet::storage::adapter::memory::Memory::default();
            }
        )
```

guess you can't use cfg-if as an expression 🤷‍♀️